incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -2073,12 +2200,72 @@ protected void doKafkaData(
             assert initialSeq <= initialAck + initialMax;
         }
 
+        protected void cancelExpirySignal(
+            long authorization,
+            long traceId)
+        {
+            String16FW expirySignalKey = new String16FW.Builder()
+                .wrap(sessionSignalKeyBuffer, 0, sessionSignalKeyBuffer.capacity())
+                .set(delegate.clientId.asString() + EXPIRY_SIGNAL_KEY_POSTFIX, StandardCharsets.UTF_8).build();
+            Flyweight expirySignalKafkaDataEx = kafkaDataExRW
+                .wrap(extBuffer, 0, extBuffer.capacity())
+                .typeId(kafkaTypeId)
+                .merged(m -> m
+                    .deferred(0)
+                    .timestamp(now().toEpochMilli())
+                    .partition(p -> p.partitionId(-1).partitionOffset(-1))
+                    .key(b -> b.length(expirySignalKey.length())
+                        .value(expirySignalKey.value(), 0, expirySignalKey.length()))
+                    .hashKey(b -> b.length(delegate.clientId.length())
+                        .value(delegate.clientId.value(), 0, delegate.clientId.length()))
+                    .headersItem(h ->
+                        h.nameLen(TYPE_HEADER_NAME.length())
+                            .name(TYPE_HEADER_NAME.value(), 0, TYPE_HEADER_NAME.length())
+                            .valueLen(EXPIRY_SIGNAL_NAME.length())
+                            .value(EXPIRY_SIGNAL_NAME.value(), 0, EXPIRY_SIGNAL_NAME.length())))
+                .build();
+
+            doKafkaData(traceId, authorization, 0, 0, DATA_FLAG_COMPLETE,
+                null, expirySignalKafkaDataEx);
+        }
+
+        protected void sendExpirySignal(
+            long authorization,
+            long traceId,
+            Flyweight payload)
+        {
+            String16FW expirySignalKey = new String16FW.Builder()
+                .wrap(sessionSignalKeyBuffer, 0, sessionSignalKeyBuffer.capacity())
+                .set(delegate.clientId.asString() + EXPIRY_SIGNAL_KEY_POSTFIX, StandardCharsets.UTF_8).build();
+            Flyweight expirySignalKafkaDataEx = kafkaDataExRW
+                .wrap(extBuffer, 0, extBuffer.capacity())
+                .typeId(kafkaTypeId)
+                .merged(m -> m
+                    .deferred(0)
+                    .timestamp(now().toEpochMilli())
```

Comment:
We have been using `System.currentTimeMillis()` everywhere, let's keep that consistent here too.
Please check for other occurrences of `Instant.now().toEpochMillis()` so they can use `System.currentTimeMillis()` instead.